bin/add_practice_exercise

```diff
@@ -30,51 +37,60 @@ command -v curl >/dev/null 2>&1 || {
     exit 1
 }
 
+# Check if configlet is already fetched
+if [ -e bin/configlet ]; then
+    message "success" "Configlet found!"
+else
+    message "info" "Fetching configlet"
+    bin/fetch-configlet
+    message "success" "Fetched configlet successfully!"
+fi
+
 # Check if exercise exists in configlet info or in config.json
 check_exercise_existence "$1"
 
 # ==================================================
 
-SLUG="$1"
-HAS_CANONICAL_DATA=true
+slug="$1"
 # Fetch canonical data
-canonical_json=$(bin/fetch_canonical_data "$SLUG")
+canonical_json=$(bin/fetch_canonical_data "$slug")
 
+has_canonical_data=true
 if [ "${canonical_json}" == "404: Not Found" ]; then
-    HAS_CANONICAL_DATA=false
+    has_canonical_data=false
     message "warning" "This exercise doesn't have canonical data"
 
 else
     echo "$canonical_json" >canonical_data.json
     message "success" "Fetched canonical data successfully!"
 fi
 
-UNDERSCORED_SLUG=$(dash_to_underscore "$SLUG")
-EXERCISE_DIR="exercises/practice/${SLUG}"
-EXERCISE_NAME=$(format_exercise_name "$SLUG")
-message "info" "Using ${YELLOW}${EXERCISE_NAME}${BLUE} as a default exercise name. You can edit this later in the config.json file"
+underscored_slug=$(dash_to_underscore "$slug")
+exercise_dir="exercises/practice/${slug}"
+exercise_name=$(format_exercise_name "$slug")
+message "info" "Using ${yellow}${exercise_name}${blue} as a default exercise name. You can edit this later in the config.json file"
 # using default value for difficulty
-EXERCISE_DIFFICULTY=$(validate_difficulty_input "${2:-$(get_exercise_difficulty)}")
-message "info" "The exercise difficulty has been set to ${YELLOW}${EXERCISE_DIFFICULTY}${BLUE}. You can edit this later in the config.json file"
+exercise_difficulty=$(validate_difficulty_input "${2:-$(get_exercise_difficulty)}")
+message "info" "The exercise difficulty has been set to ${yellow}${exercise_difficulty}${blue}. You can edit this later in the config.json file"
 # using default value for author
-AUTHOR_HANDLE=${3:-$(get_author_handle)}
-message "info" "Using ${YELLOW}${AUTHOR_HANDLE}${BLUE} as author's handle. You can edit this later in the 'authors' field in the ${EXERCISE_DIR}/.meta/config.json file"
-
-
-create_rust_files "$EXERCISE_DIR" "$SLUG" "$HAS_CANONICAL_DATA"
+author_handle=${3:-$(get_author_handle)}
+message "info" "Using ${yellow}${author_handle}${blue} as author's handle. You can edit this later in the 'authors' field in the ${exercise_dir}/.meta/config.json file"
 
+create_rust_files "$exercise_dir" "$slug" "$has_canonical_data"
 
 # ==================================================
 
-# build configlet
+# Build configlet
 ./bin/fetch-configlet
```

Comment:
This is trying to fetch configlet a second time. I guess this is no longer needed?